### DISCUSSION TITLE
Flipping/Mirroring of Layers

### DISCUSSION THREAD

**Discussion by francescomartinod** [openlayers/openlayers#15917] [Q&A]
Hey All,
I am trying to achieve the flipping of a map over X (or Y) axis but I could not find a way. 
A tried with ```addCoordinateTransforms``` and it works fine with single images, but did not manage to make it work with a TileImage. Inverting the mapping in TileUrl function, flipping the tile before rendering, resulted in strange behaviors. I am using a WebGlTileLayer
Is there any way to achieve image flipping?

**Comment by mike-000** [comment]
`addCoordinateTransforms` alone should be sufficient (for a local tile grid the transforms may be more complex than simply inverting the values).  This triggers a reprojection of the source and needs the `crossOrigin` option to be set when using WebGlTileLayer.

```
import Map from 'ol/Map.js';
import {
  Projection,
  addCoordinateTransforms,
  getPointResolution,
  get as getProjection,
} from 'ol/proj.js';
import TileLayer from 'ol/layer/WebGLTile.js';
import View from 'ol/View.js';
import XYZ from 'ol/source/XYZ.js';

const proj3857 = getProjection('EPSG:3857');

const projFlip = new Projection({
  code: 'EPSG:3857:FLIP',
  units: proj3857.getUnits(),
  extent: proj3857.getExtent(),
  global: proj3857.isGlobal(),
  getPointResolution: (resolution, point) =>
    getPointResolution(proj3857, resolution, point),
});

addCoordinateTransforms(
  proj3857,
  projFlip,
  ([x, y]) => [x, -y],
  ([x, y]) => [x, -y],
);

const map = new Map({
  layers: [
    new TileLayer({
      source: new XYZ({
        url: 'https://tile.openstreetmap.org/{z}/{x}/{y}.png',
        crossOrigin: '',
        projection: projFlip,
      }),
    }),
  ],
  target: 'map',
  view: new View({
    center: [0, 0],
    zoom: 2,
  }),
});

```
